### PR TITLE
soc: stm32f4: define uart7's dts

### PR DESCRIPTION
**Pull request by jli157**
Complete definitions for UART7 on
STM32F4 SoCs.

Signed-off-by: Jun Li <jun.r.li@intel.com>

### CODE REVIEW

File: dts/arm/st/f4/stm32f4-pinctrl.dtsi [hunk 1]
```diff
@@ -69,6 +69,18 @@
 					tx = <STM32_PIN_PG14 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_NOPULL)>;
 				};
 			};
+			uart7_pins_a: uart7_a {
+				rx_tx {
+					rx = <STM32_PIN_PE8 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUPDR_NO_PULL)>;
+					tx = <STM32_PIN_PE7 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_NOPULL)>;
+				};
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4-pinctrl.dtsi]: From the datasheet I have before my eyes:
PE8 > TX
PE7 > RX

File: dts/arm/st/f4/stm32f4-pinctrl.dtsi [hunk 2]
```diff
@@ -69,6 +69,18 @@
...
+			};
+			uart7_pins_b: uart7_b {
+				rx_tx {
+					rx = <STM32_PIN_PF7 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUPDR_NO_PULL)>;
+					tx = <STM32_PIN_PF6 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_NOPULL)>;
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4-pinctrl.dtsi]: Ditto

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by jli157** [comment]
@erwango fixed. Sorry tx-rx order was haunting in my mind. :)

**Comment by erwango** [comment]
> @erwango fixed. Sorry tx-rx order was haunting in my mind. :)

I usually get confused as well. This is why I pay attention to that in reviews.
Maybe we should reverse tx/rx order :-)

**Review by erwango — APPROVED** [review]

**Pull request merged**